compiler-cli/src/run.rs

```diff
@@ -90,9 +90,33 @@ fn run_javascript(
 
     // Tell Node that any following argument are for the program
     args.push("--".into());
-    for argument in arguments.into_iter() {
-        args.push(argument);
-    }
+    args.append(&mut arguments);
+
+    ProjectIO::new().exec("node", None, &args, &[], None, false)
+}
+
+fn run_deno(
+    config: &PackageConfig,
+    module: &str,
+    mut arguments: Vec<String>,
+) -> Result<i32, Error> {
+    let mut args = vec![
+        "run".to_owned(),
+        "-A".to_owned(), // Allow all permissions
+    ];
+
+    let module = paths::build_package(Mode::Dev, Target::JavaScript, &config.name)
+        .join("dist")
+        .join(module);
+
+    // Run the main function.
+    args.push("-".into());
+    let script = format!(
+        "import('./{}.mjs').then(module => module.main())",
+        module.to_string_lossy()
+    );
+
+    args.append(&mut arguments);
 
-    ProjectIO::new().exec("node", &args, &[], None, false)
+    ProjectIO::new().exec("deno", Some(script.as_bytes()), &args, &[], None, false)
```

Comment:
Can we pass this program as an argument? I imagine in future we will want this to work `cat somefile > gleam run` so we'll need to be able to use stdin from the program running on Deno

Reply:
We could pass it as an argument if we emitted the bit of code as a file (which I think is a good idea anyway), but I couldn’t find the right spot to do so. 

If we go that route, I feel like we should probably only emit it if the appropriate module actually has a `main` function to run, which I have no idea how to check.

Reply:
There's a file generated for Erlang called `erlang@@main` or something along those lines, we could probably copy how that is created.

> I feel like we should probably only emit it if the appropriate module actually has a main function to run, which I have no idea how to check.

Let's always emit it but have an issue to add a nice error if we detect that the main function is missing.